packages/report-server/src/reportBuilder/transform/parser/TransformParser.ts

```diff
@@ -65,14 +77,22 @@ export class TransformParser extends ExpressionParser {
       });
       this.set('where', this.whereFunction); // no '@' prefix for where
     }
+
+    this.context = context;
   }
 
-  public evaluate(expression: string) {
-    if (TransformParser.isExpression(expression)) {
-      return super.evaluate(expression.substring(1));
-    }
+  public isExpression(input: unknown) {
```

Comment:
I know I'm being nitpicky here, but mind if we make this `static` again? Given that this function independnant of the instance state, it makes sense for it to be static to me? It also means that we can check if a input is an expression without instantiating a `TransformParser` (we do this in `sortRows` for instance)

Reply:
Now that `EXPRESSION_PREFIX` is not inherited from the base class, we could. I'll do that.

Disclaimer: I generally avoid static methods unless if there is a very specific reason to use one, since they are [hard to mock](https://testing.googleblog.com/2008/12/static-methods-are-death-to-testability.html) + don't fit right in an inheritance model. Of course, as with almost everything, there are justifiable use cases for them!

(The article I linked has a very ominous, click-baity title 😛 But I have also confirmed that static methods are hard to mock, at least with `sinon` )

Reply:
Having said that, I've noticed in the past that instantiating `ExpressionParser` took much more than one would expect (something like `30ms`). In fact, they accounted for 70% of the slow performance in the initial implementation of [nested indicators](https://github.com/beyondessential/tupaia/pull/1835)!

I don't know if this is still the case, but I am flagging it since `TransformParser` is instantiated multiple times in `report-server`. If it is, a simple solution around that would be using a singleton.

Reply:
I agree that it's good to be wary about static methods. You really need to be careful that they don't mutate state, and yes they are very hard to mock. That being said, in this instance, and in general, I do find them appropriate for places where they perform simple, functional logic, and that mocking wouldn't be appropriate.

For instance you could in theory mock `isExpression` to always return true when testing various expressions in report-server, to save yourself the trouble of putting `=` at the start of them. But I think that would be a misuse of mocking. Leaving it as is and adding the `=` improves your tests as it aligns your tests with how the code actually operates.

Reply:
As for the performance concerns around instantiating `TransformParser`. I share them too, but I think for now it's acceptable. We insatiate at once per transform I believe, which could be up to 10 or so? Singleton pattern could work but would require a bit more overhead of cleaning up scope between uses...

Reply:
10 times 30 is 300ms, a significant overhead for something that doesn't access the file system...